### PR TITLE
Thread generator through trunc_normal_

### PR DESCRIPTION
**Pull request by tfsingh**
This will solve @albertz's issue as described in #98200 , threading the generator argument through the trunc_normal_ function. I'm still working on #99796 (and won't let it stall out), but this fix doesn't trigger any JIT issues, so I think it might be helpful to get it merged now.

Would be happy to iterate on this if there are any issues.

### CODE REVIEW

**Review by Skylion007 — APPROVED** [review]

**Review by albanD — COMMENTED** [review]
Sounds good!
Can you add a test for this please in test/nn/test_init.py ?

**Comment by tfsingh** [comment]
Of course, just added one

File: test/nn/test_init.py [hunk 1]
```diff
@@ -121,6 +121,15 @@ def test_trunc_normal(self):
 
             assert self._is_trunc_normal(input_tensor, mean, std, a, b)
 
+    def test_trunc_normal_generator(self):
+        gen = torch.Generator()
+        gen.manual_seed(42)
+        input_tensor = torch.empty(5)
+        init.trunc_normal_(input_tensor, generator=gen)
+        expected = torch.tensor([1.1025, 1.2598, -0.2840, 1.5415, -0.2652])
```
Comment (albanD) [inline test/nn/test_init.py:129]: Nothing gurantees that the rng won't change across versions.
You can use `self._is_trunc_normal()` on the generated Tensor to ensure it was properly generated (see test above for details).

You also want to make sure the generator is being used properly, you can add:
```python
ref = torch.empty(5)
torch.manual_seed(42)
init.trun_normal_(ref)

self.assertEqual(input_tensor, ref)
```

Comment (tfsingh) [reply test/nn/test_init.py]: Just fixed this -- my apologies as I didn't thoroughly check how self._is_trunc_normal() was implemented (I incorrectly assumed it used an equivalent method in scipy, and I wasn't sure if scipy had generator support). I'll be sure to take a closer look at surrounding tests in the future, hope this is good now!

**Review by albanD — APPROVED** [review]
Thanks!

File: test/nn/test_init.py (hunk 1 above)
Comment (albanD) [reply test/nn/test_init.py]: No worries. Thanks for the quick update!

**Comment by albanD** [comment]
Github is in a bad state. Let's wait a bit!

**Pull request closed without merging**